src/adapters/google-maps.adapter.spec.ts

```diff
@@ -64,4 +80,938 @@ describe("TerraDrawGoogleMapsAdapter", () => {
+			}));
+
+			const adapter = new TerraDrawGoogleMapsAdapter({
+				lib: {
+					LatLng: jest.fn(),
+					OverlayView: jest.fn().mockImplementation(() => ({
+						setMap: jest.fn(),
+						getProjection: getProjectionMock,
+					})),
+					Point: jest.fn(),
+				} as any,
+				map: createMockGoogleMap(),
+			});
+
+			const unprojected = adapter.unproject(50, 80);
+			expect(getProjectionMock).toHaveBeenCalledTimes(1);
+
+			expect(unprojected.lng).toEqual(testLng);
+			expect(unprojected.lat).toEqual(testLat);
+		});
+	});
+
+	describe("setCursor", () => {
+		it("is no-op for cursor: unset", () => {
+			const map = createMockGoogleMap() as google.maps.Map;
+			const container = {
+				offsetLeft: 0,
+				offsetTop: 0,
+				style: { removeProperty: jest.fn(), cursor: "initial" },
+			} as any;
+
+			map.getDiv = jest.fn(() => container);
+
+			// Create the adapter instance with the mocked map
+			const adapter = new TerraDrawGoogleMapsAdapter({
+				lib: {
+					OverlayView: jest.fn().mockImplementation(() => ({
+						setMap: jest.fn(),
+						getProjection: jest.fn(),
+					})),
+				} as any,
+				map,
+			});
+
+			adapter.setCursor("unset");
+
+			expect(map.getDiv).not.toHaveBeenCalled();
+			expect(container.style.removeProperty).not.toHaveBeenCalledTimes(1);
+		});
+
+		it("sets to pointer", () => {
+			const elId = "map-container";
+			const map = createMockGoogleMap() as google.maps.Map;
+			const container = {
+				...document.createElement("div"),
+				offsetLeft: 0,
+				offsetTop: 0,
+				id: elId,
+			};
+
+			map.getDiv = jest.fn(() => container);
+
+			const adapter = new TerraDrawGoogleMapsAdapter({
+				lib: {
+					OverlayView: jest.fn().mockImplementation(() => ({
+						setMap: jest.fn(),
+						getProjection: jest.fn(),
+					})),
+				} as any,
+				map,
+			});
+
+			adapter.setCursor("pointer");
+
+			expect(map.getDiv).toHaveBeenCalledTimes(1);
+			const firstSheetAndRule = document.styleSheets[0]
+				.cssRules[0] as CSSStyleRule;
+			expect(
+				firstSheetAndRule.selectorText.startsWith(`#${elId}`),
+			).toBeTruthy();
+			expect(
+				firstSheetAndRule.cssText.includes(`cursor: pointer`),
+			).toBeTruthy();
+		});
+
+		it("exits early when no update to cursor type", () => {
+			const elId = "map-container";
+			const map = createMockGoogleMap() as google.maps.Map;
+			const container = {
+				...document.createElement("div"),
+				offsetLeft: 0,
+				offsetTop: 0,
+				id: elId,
+			};
+
+			map.getDiv = jest.fn(() => container);
+
+			const adapter = new TerraDrawGoogleMapsAdapter({
+				lib: {
+					OverlayView: jest.fn().mockImplementation(() => ({
+						setMap: jest.fn(),
+						getProjection: jest.fn(),
+					})),
+				} as any,
+				map,
+			});
+
+			adapter.setCursor("pointer");
+			adapter.setCursor("pointer");
+			adapter.setCursor("pointer");
+
+			expect(map.getDiv).toHaveBeenCalledTimes(1);
+		});
+	});
+
+	describe("render", () => {
+		it("first invocation registers event listeners", () => {
```

Comment:
I was actually slightly surprised that `render` seemed to take on an 'initialization' role here (registering event listeners the first time it is called). The ArcGIS adapter, for instance, does this via a separate method, [invoked in the constructor](https://github.com/JamesLMilner/terra-draw/blob/main/src/adapters/arcgis-maps-sdk.adapter.ts#L59). Would you be open to refactoring changes to move the 'init' code to a separate method, and invoking in the constructor ?

Reply:
Yes this is a fair observation and you're probably right that these would be better suited in the constructor. I think lets look into as a separate issue as not to block your hard work?

Reply:
yes, was thinking of a separate set of changes